docker/recorder/src/main/java/org/arquillian/cube/docker/impl/client/reporter/TakeDockerEnvironment.java

```diff
@@ -58,9 +75,66 @@ public void reportDockerEnvironment(@Observes AfterAutoStart event, CubeDockerCo
         docker.getPropertyEntries().add(containersComposition);
 
         propertyReportEvent.fire(new PropertyReportEvent(docker));
+    }
+
+    public void reportContainerStatsBeforeTest(@Observes Before before, DockerClientExecutor executor) {
+        generateStats(executor, false, "Before");
+    }
+
+    public void reportContainerStatsAfterTest(@Observes After after, DockerClientExecutor executor, ReporterConfiguration configuration) {
+        generateStats(executor, false, "After");
+    }
+
+    private void generateStats(DockerClientExecutor executor, Boolean decimal, String when) {
+        if (executor != null) {
+            List<Cube<?>> containers = cubeRegistryInstance.get().getCubes();
 
+            GroupEntry containersStat = new GroupEntry("Container Statistics " + when + " Method");
+
+            for (Cube<?> container : containers) {
+                String name = container.getId();
+                Statistics statistics = executor.statsContainer(name);
+                Map<String, Map> stats = getStats(statistics, decimal);
+                containersStat.getPropertyEntries().add(createContainerStatGroup(stats, name));
+            }
+            propertyReportEvent.fire(new PropertyReportEvent(containersStat));
+        }
+    }
+
+    private GroupEntry createContainerStatGroup(Map<String, Map> stats, String id) {
+
+        GroupEntry containerStatsInfo = new GroupEntry(id + " Statistics");
+
+        for (Map.Entry<String, Map> stat : stats.entrySet()) {
+            GroupEntry info = new GroupEntry(stat.getKey() + " statistics");
+            for(Object entry : stat.getValue().entrySet()){
+                Map.Entry castEntry = (Map.Entry) entry;
+                addEntry(new KeyValueEntry(castEntry.getKey().toString(),castEntry.getValue().toString()), info);
+            }
+            containerStatsInfo.getPropertyEntries().add(info);
+         }
+
+        return containerStatsInfo;
     }
 
+   public void reportLogs(@Observes org.arquillian.cube.spi.event.lifecycle.BeforeStop beforeStop, DockerClientExecutor executor, ReporterConfiguration reporterConfiguration) {
+        final String cubeId = beforeStop.getCubeId();
+        if (cubeId != null) {
+            final File logFile = new File(reporterConfiguration.getRootDir(), cubeId + ".log");
+            try {
+                executor.copyLog(beforeStop.getCubeId(), false, true, true, false, -1, new FileOutputStream(logFile));
```

Comment:
@lordofthejars it looks like executor API needs some love. Lots of magic values here